tools/generator/templates/player/haskell/Makefile-haskell.jinja2

```diff
@@ -0,0 +1,38 @@
+{# SPDX-License-Identifier: GPL-2.0-or-later
+# Copyright (c) 2020 Association Prologin <association@prologin.org>
+#}
+{% extends "includes/Makefile-language.jinja2" %}
+
+{% block lang %}haskell{% endblock %}
+
+{% block extra_flags %}
+GHC    ?= ghc
+HSC2HS ?= hsc2hs
+
+GHC_FLAGS = -pgml $(CXX) -O9 -dynamic --make -shared -fPIC \
+			-L$(shell $(GHC) --print-libdir) \
+			-lHSrts-ghc$(shell $(GHC) --numeric-version)
+
+CXXFLAGS += -O2 -I$(shell $(GHC) --print-libdir)/include
```

Comment:
I think `-O2` should be in the default CXXFLAGS instead of being defined here, what do you think?